exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -607,7 +607,14 @@ public long getRowCount() {
       fileSet = Sets.newHashSet(fileNames);
     }
 
-    if (fileNames.isEmpty()) {
+    List<String> finalFileNames;
+    if (fileSet != null) {
+      finalFileNames = Lists.newArrayList(fileSet);
```

Comment:
Yes, I can try to keep only the fileSet and get rid of fileNames (although I would still need the finalFileNames list since FileSelection only accepts a List, not a Set).